packages/fluentui/react-northstar-prototypes/src/prototypes/VirtualizedStickyTree/VirtualStickyTree.tsx

```diff
@@ -0,0 +1,368 @@
+  TreeProps,
+  Tree,
+  TreeStylesProps,
+  treeClassName,
+  useVirtualTree,
+  GetItemById,
+} from '@fluentui/react-northstar';
+import { TreeContext, TreeRenderContextValue } from '@fluentui/react-northstar/src/components/Tree/context';
+import { VariableSizeList, VariableSizeListProps, ListChildComponentProps } from 'react-window';
+
+export interface VirtualStickyTreeProps
+  extends Omit<TreeProps, 'selectedItemIds' | 'defaultSelectedItemIds' | 'onSelectedItemIdsChange' | 'selectable'>,
+    Pick<VariableSizeListProps, 'height'> {
+  /** height of a non-sticky tree item */
+  itemSize: number;
+  /** height of 1st level sticky tree item */
+  stickyItemSize: number;
+}
+
+export interface InnerElementContextType {
+  getItemById: GetItemById;
+  stickyItemIds: string[];
+  stickyItemPusherHeights: number[];
+  stickyItemSize: number;
+}
+
+export interface VirtualItemData {
+  visibleItemIds: string[];
+  createTreeItem: (id: string, style: React.CSSProperties) => React.ReactElement<TreeItemProps> | null;
+}
+
+export const InnerElementContext = React.createContext<InnerElementContextType>({} as InnerElementContextType);
+
+export const VirtualStickyTreeClassName = 'ui-virtualstickytree';
+
+export const VirtualStickyTree: ComponentWithAs<'div', VirtualStickyTreeProps> = props => {
+  const context = useFluentContext();
+
+  const { children, className, design, styles, variables, items, height, itemSize, stickyItemSize } = props;
+
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps([...Tree.handledProps, 'stickyItemSize', 'itemSize'], props);
+
+  const getA11yProps = useAccessibility(props.accessibility, {
+    debugName: VirtualStickyTree.displayName,
+    rtl: context.rtl,
+  });
+
+  const { classes } = useStyles<TreeStylesProps>(Tree.displayName, {
+    className: treeClassName,
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
+
+  const stickyItemIds = React.useMemo(() => items.map(item => item.id), [items]);
+
+  const {
+    visibleItemIds,
+    getItemById,
+    registerItemRef,
+    activeItemIds,
+    toggleItemActive: baseToggleItemActive,
+    focusItemById,
+    expandSiblings,
+    listRef,
+    getItemRef,
+  } = useVirtualTree({ ...props, defaultActiveItemIds: stickyItemIds });
+
+  // get height of the pusher for each sticky item, based on number of their visible descendants
+  const stickyItemPusherHeights: number[] = React.useMemo(() => {
+    const result = new Array(stickyItemIds.length).fill(0);
+
+    let stickyIndex = 0;
+    visibleItemIds.forEach(id => {
+      if (id === stickyItemIds[stickyIndex]) {
+        stickyIndex++;
+      } else {
+        result[stickyIndex - 1] += itemSize;
+      }
+    });
+    return result;
+  }, [itemSize, stickyItemIds, visibleItemIds]);
+
+  const toggleItemActive = React.useCallback(
+    (e: React.SyntheticEvent, idToToggle: string) => {
```

Comment:
nit: would this be more appropriate ?

```suggestion
    (e: React.MouseEvent | React.KeyboardEvent, idToToggle: string) => {
```

Reply:
Maybe, but I think I'd keep it here because `useTree` hook has also `toggleItemActive` and I'd like to have the same type